### DISCUSSION TITLE
Is it possible to customize the `display` utilities? I'd like to rename `.hidden`

### DISCUSSION THREAD

**Discussion by MichaelAllenWarner** [tailwindlabs/tailwindcss#12941] [Help] [answered]
Is it possible to rename the `.hidden` utility? Couldn't find a way to customize the core `display` plugin.

I've got a situation where some non-negotiable 3rd-party CSS is coming in with an un-scoped rule like `.hidden { display: none !important }`. This prevents me from doing something like `hidden sm:block` in my Tailwind, and since the 3rd-party CSS necessarily ends up appended to the _bottom_ of `<head>` (because it's put there by 3rd-party JS that itself is fetched dynamically and thus uneditable), even `hidden sm:!block` wouldn't get around the problem.

My best option at the moment is probably to make a new utility like `.display-none` that does the same thing as `.hidden`, and then replace all my `.hidden` instances with it. That's manageable. But it would be even better if I could also _disable_ the `hidden` utility, so that I and other developers don't accidentally use it (and also so that my new utility appears alongside the other `display` utilities in the resulting CSS).

**Comment by wongjn** [comment, marked as answer]
You could consider using [`blocklist` configuration option to prevent Tailwind from ever generating the CSS rule for `hidden`](https://tailwindcss.com/docs/content-configuration#discarding-classes):

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  blocklist: [
    'hidden',
  ],
  // …
}
```
Then, you can create your own utility class for `display: none` [in CSS](https://tailwindcss.com/docs/adding-custom-styles#adding-custom-utilities) or [in a Tailwind Plugin](https://tailwindcss.com/docs/plugins#static-utilities).

**Reply by MichaelAllenWarner** [reply]
Yep, I think this is probably the way to go. Ideally I could add the new class as a `display` utility (for CSS co-location), but disabling with the `blocklist` (which I'd completely forgotten about) will almost certainly suffice. Thanks!

**Comment by jaimeenkoffeekodes** [comment]
Yes, it is possible to rename the `.hidden` utility in Tailwind CSS. You can achieve this by customizing the configuration and disabling the built-in `.hidden` utility, then creating your own utility with the desired name and styles.

Here's how you can do it:

1. Disable the built-in `.hidden` utility in your Tailwind CSS configuration:


// tailwind.config.js

module.exports = {
  // other configuration options
  corePlugins: {
    // Disable the built-in hidden utility
    display: false,
  },
}


2. Define your own utility with the desired name and styles:

// tailwind.config.js

module.exports = {
  // other configuration options
  theme: {
    extend: {
      // Define your custom utility with the desired name and styles
      display: {
        none: 'none',
      },
    },
  },
  plugins: [
    // Optionally, you can add your own custom plugin here
  ],
}

Now you can use the `.display-none` utility instead of `.hidden` in your codebase. This approach ensures that you don't accidentally use the built-in `.hidden` utility and allows you to use your own custom utility alongside other Tailwind CSS utilities.

**Reply by MichaelAllenWarner** [reply]
Thank you for your response. Unfortunately, your suggestion doesn't work.

First: disabling the core `display` plugin means sacrificing utilities like `flex` and `grid`. (I suppose that one could then reproduce them all by hand, but that seems heavy-handed.)

Second: I haven't had success attempting to `extend` the `display` utilities. It simply doesn't work!

**Reply by wongjn** [reply]
Indeed, this answer is wholely incorrect.
